### PR TITLE
Added static migration proposal store

### PR DESCRIPTION
**Pull request by cwh42**
In respect to your comments in https://github.com/yast/yast-migration/pull/4

### CODE REVIEW

File: src/lib/migration/proposal_store.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "installation/proposal_order"
```
Comment (jreidinger) [inline src/lib/migration/proposal_store.rb]: typo, it should be proposal_store which you use below.

File: src/lib/migration/proposal_store.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+module Migration
+  class ProposalStore < Installation::ProposalStore
+    include Yast::Logger
+    include Yast::I18n
+
+    PROPOSAL_NAMES = [ "one", "two", "three" ]
```
Comment (jreidinger) [inline src/lib/migration/proposal_store.rb]: please use data I send ` [ "update_proposal"]` to have it also please require yast2-update where client lives.

File: src/lib/migration/proposal_store.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    PROPOSAL_PROPERTIES = { "key" => "value" }
```
Comment (jreidinger) [inline src/lib/migration/proposal_store.rb]: I think for now it can be empty hash

File: src/lib/migration/proposal_store.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    MODULES_ORDER = [ "one", "two", "three" ]
```
Comment (jreidinger) [inline src/lib/migration/proposal_store.rb]: For now I think it can be `MODULES_ORDER = PROPOSAL_NAMES`

**Comment by jreidinger** [comment]
otherwise it looks good, just please add also test cases

File: src/lib/migration/proposal_store.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+module Migration
+  # Provides access to static metadata for update proposal.
+  class ProposalStore < Installation::ProposalStore
+    include Yast::Logger
+    include Yast::I18n
+
+    PROPOSAL_NAMES = ["update_proposal"]
+    PROPOSAL_PROPERTIES = {}
+    MODULES_ORDER = PROPOSAL_NAMES
+
```
Comment (jreidinger) [inline src/lib/migration/proposal_store.rb]: missing textdomain for translation. Also it have to have initialization with argument due to way how it is initialized so something like:
```
def initialize(_unused_mode)
  textdomain "migration"
 
  super("migration")
end
```

File: test/proposal_store_test.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require_relative "test_helper"
+
+require "migration/proposal_store"
+
+describe Migration::ProposalStore do
+  ps = Migration::ProposalStore.new(nil)
```
Comment (jreidinger) [inline test/proposal_store_test.rb]: more rspec style is `subject { Migration::ProposalStore.new(nil) }` and then use in text `subject` if you prefer ps you can use `subject(:ps) { .. }`

File: test/proposal_store_test.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    it "returns an update proposal" do
+      expect(ps.proposal_names).to eq ["update_proposal"]
+    end
+  end
+
+  describe ".presentation_order" do
+    it "returns a presentation order" do
+      expect(ps.presentation_order).to eq ["update_proposal"]
+    end
+  end
```
Comment (jreidinger) [inline test/proposal_store_test.rb]: well, private method is not tested here...it means there should be also tests for public methods from `::Installation::ProposalStore` that changed due to different private method implementation.

File: src/clients/migration_proposals.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+require "migration/proposal_store"
+require "installation/proposal_runner"
+require "yast"
+
+Yast.import "Wizard"
+
+# Proposal runner expect already opened wizard dialog
+Yast::Wizard.OpenNextBackDialog
```
Comment (lslezak) [inline src/clients/migration_proposals.rb]: Um, a new dialog is open here, but I cannot see anything like `Wizard.CloseDialog` here, looks like UI leak for me...

Comment (jreidinger) [inline src/clients/migration_proposals.rb]: yep, let me try how it will perform then

**Comment by lslezak** [comment]
LGTM (except the dialog issue)

And the 92% code coverage looks great :+1: , keep that!

**Pull request merged**